File: NetFramework/Gauge/VB/GaugeDemo/PageLituus.vb
Imports C1.Win.Gauge

Public Class PageLituus

    Public Sub New()
        MyBase.New()

        InitializeComponent()
        FillDecorators()

    End Sub

    Private Sub FillDecorators()
        Dim x As Double = 0.0, y As Double = 0.0
        Dim x1 As Double = 0.0, y1 As Double = 0.0

        c1Gauge1.BeginUpdate()

        Dim cm As CommonColorMap = New CommonColorMap()
        cm.Name = "green"
        cm.ValueColorFalloff = C1GaugeValueColorFalloff.None
        cm.ValueColors.Add(New C1GaugeValueColor(0.0, Color.LimeGreen, 1.0))
        cm.ValueColors.Add(New C1GaugeValueColor(Double.NaN, 100, Color.DimGray, 1.0))
        c1LinearGauge1.ColorMaps.Add(cm)

        cm = New CommonColorMap()
        cm.Name = "yellow"
        cm.ValueColorFalloff = C1GaugeValueColorFalloff.None
        cm.ValueColors.Add(New C1GaugeValueColor(0.0, Color.Yellow, 1.0))
        cm.ValueColors.Add(New C1GaugeValueColor(Double.NaN, 100, Color.DimGray, 1.0))
        c1LinearGauge1.ColorMaps.Add(cm)

        cm = New CommonColorMap()
        cm.Name = "red"
        cm.ValueColorFalloff = C1GaugeValueColorFalloff.None
        cm.ValueColors.Add(New C1GaugeValueColor(0.0, Color.FromArgb(255, 64, 64), 1.0))
        cm.ValueColors.Add(New C1GaugeValueColor(Double.NaN, 100, Color.DimGray, 1.0))
        c1LinearGauge1.ColorMaps.Add(cm)

        For i As Int32 = 0 To 99
            Dim mark As C1GaugeSingleMark = New C1GaugeSingleMark()
            mark.Border.LineStyle = C1GaugeBorderStyle.None
            mark.Shape = C1GaugeMarkShape.Round
            mark.Width = 2
            Dim dl As Double = i * 0.035
            mark.Length = 6 + dl * dl * dl
            If (i < 60) Then
                mark.ColorMapName = "green"
            ElseIf (i < 85) Then
                mark.ColorMapName = "yellow"
            Else
                mark.ColorMapName = "red"
            End If
            mark.Value = i / 100.0

            GetXY(i, x, y)
            mark.Position = x * 0.01 + 0.27
            mark.Location = y + 35

            GetXY(i + 0.01, x1, y1)
            Dim dy As Double = (y1 - y) * 0.75
            Dim l As Double = Math.Sqrt((x1 - x) * (x1 - x) + dy * dy)
            Dim alpha As Double = Math.Asin(dy / l)
            If x1 < x Then
                alpha = Math.PI - alpha
            End If
            mark.ShapeAngle = alpha * 180.0 / Math.PI

            c1LinearGauge1.Decorators.Add(mark)
        Next i
        c1Gauge1.EndUpdate()
    End Sub

    Private Sub GetXY(ByVal a As Double, ByRef x As Double, ByRef y As Double)
        a = 120.0 - a
        Dim phi As Double = a * a * Math.PI / 5000.0
        Dim ro As Double = 36.0 / Math.Sqrt(phi)
        x = Math.Cos(phi) * ro
        y = Math.Sin(phi) * ro * 1.5
    End Sub

    Private Sub trackBar1_ValueChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles trackBar1.ValueChanged
        c1LinearGauge1.Value = trackBar1.Value / 100.0
    End Sub
End Class

File: NetFramework/Gauge/CS/GaugeDemo/PageLituus.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Text;
using System.Windows.Forms;
using C1.Win.Gauge;

namespace GaugeDemo
{
    public partial class PageLituus : BaseGaugePage
    {
        public PageLituus()
        {
            InitializeComponent();
            FillDecorators();
        }

        private void FillDecorators()
        {
            c1Gauge1.BeginUpdate();

            CommonColorMap colorMap = new CommonColorMap();
            colorMap.Name = "green";
            colorMap.ValueColorFalloff = C1GaugeValueColorFalloff.None;
            colorMap.ValueColors.Add(new C1GaugeValueColor(0.0, Color.LimeGreen, 1.0));
            colorMap.ValueColors.Add(new C1GaugeValueColor(double.NaN, 100, Color.DimGray, 1.0));
            c1LinearGauge1.ColorMaps.Add(colorMap);

            colorMap = new CommonColorMap();
            colorMap.Name = "yellow";
            colorMap.ValueColorFalloff = C1GaugeValueColorFalloff.None;
            colorMap.ValueColors.Add(new C1GaugeValueColor(0.0, Color.Yellow, 1.0));
            colorMap.ValueColors.Add(new C1GaugeValueColor(double.NaN, 100, Color.DimGray, 1.0));
            c1LinearGauge1.ColorMaps.Add(colorMap);

            colorMap = new CommonColorMap();
            colorMap.Name = "red";
            colorMap.ValueColorFalloff = C1GaugeValueColorFalloff.None;
            colorMap.ValueColors.Add(new C1GaugeValueColor(0.0, Color.FromArgb(255, 64, 64), 1.0));
            colorMap.ValueColors.Add(new C1GaugeValueColor(double.NaN, 100, Color.DimGray, 1.0));
            c1LinearGauge1.ColorMaps.Add(colorMap);

            for (int i = 0; i < 100; i++)
            {
                C1GaugeSingleMark mark = new C1GaugeSingleMark();
                mark.Border.LineStyle = C1GaugeBorderStyle.None;
                mark.Shape = C1GaugeMarkShape.Round;
                mark.Width = 2;
                double dl = i * 0.035;
                mark.Length = 6 + dl * dl * dl;
                if (i < 60)
                    mark.ColorMapName = "green";
                else if (i < 85)
                    mark.ColorMapName = "yellow";
                else
                    mark.ColorMapName = "red";
                mark.Value = i / 100.0;

                double x, y;
                GetXY(i, out x, out y);
                mark.Position = x * 0.01 + 0.27;
                mark.Location = y + 35;

                double x1, y1;
                GetXY(i + 0.01, out x1, out y1);
                double dy = (y1 - y) * 0.75;
                double l = Math.Sqrt((x1 - x) * (x1 - x) + dy * dy);
                double alpha = Math.Asin(dy / l);
                if (x1 < x)
                {
                    alpha = Math.PI - alpha;
                }
                mark.ShapeAngle = alpha * 180.0 / Math.PI;

                c1LinearGauge1.Decorators.Add(mark);
            }
            c1Gauge1.EndUpdate();
        }

        private static void GetXY(double a, out double x, out double y)
        {
            a = 120.0 - a;
            double phi = a * a * Math.PI / 5000.0;
            double ro = 36.0 / Math.Sqrt(phi);
            x = Math.Cos(phi) * ro;
            y = Math.Sin(phi) * ro * 1.5;
        }

        private void trackBar1_ValueChanged(object sender, EventArgs e)
        {
            c1LinearGauge1.Value = trackBar1.Value / 100.0;
        }
    }
}
